app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dec = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        key = (*args, *kwargs.items())
+
+        if key in cache_dec:
+            print("Getting from cache")
+            return cache_dec[key]
+        else:
+            print("Calculating new result")
```

Comment:
```suggestion

        print("Calculating new result")
```
else: redundant in this case